lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -76,36 +99,92 @@ public function get_providers() {
 	/**
 	 * Register a webfont.
 	 *
-	 * @param array $font The font arguments.
+	 * @param array $webfont The font argument.
 	 */
-	public function register_font( $font ) {
-		$font = $this->validate_font( $font );
-		if ( $font ) {
-			$id                    = $this->get_font_id( $font );
-			$this->webfonts[ $id ] = $font;
+	public function register_webfont( $webfont ) {
+		$webfont = $this->validate_webfont( $webfont );
+
+		// If not valid, bail out.
+		if ( ! $webfont ) {
+			return false;
 		}
+
+		$slug = $this->get_font_slug( $webfont );
+
+		// Initialize a new font-family collection.
+		if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+			$this->registered_webfonts[ $slug ] = array();
+		}
+
+		$this->registered_webfonts[ $slug ][] = $webfont;
 	}
 
 	/**
-	 * Get the font ID.
+	 * Enqueue a webfont.
+	 *
+	 * If a string is passed, it enqueues a font-family that has been already registered.
+	 * If an array is passed, it registers and enqueues a font-family.
 	 *
-	 * @param array $font The font arguments.
-	 * @return string
+	 * @param string|array $webfont The webfont to be enqueued.
 	 */
-	public function get_font_id( $font ) {
-		return sanitize_title( "{$font['font-family']}-{$font['font-weight']}-{$font['font-style']}-{$font['provider']}" );
+	public function enqueue_webfont( $webfont ) {
+		$slug = $this->get_font_slug( $webfont );
+
+		if ( isset( $this->enqueued_webfonts[ $slug ] ) ) {
+			trigger_error(
+				sprintf(
+					/* translators: %s unique slug to identify the webfont */
+					__( 'The "%s" font family is already enqueued.', 'gutenberg' ),
+					$slug
+				)
+			);
+
+			return false;
+		}
+
+		if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+			if ( is_string( $webfont ) ) {
+				_doing_it_wrong( __FUNCTION__, sprintf( __( 'The "%s" font family is not registered.' ), $slug ), '6.0.0' );
+				return false;
+			}
+
+			$this->register_webfont( $webfont );
+		}
+
+		$this->enqueued_webfonts[ $slug ] = $this->registered_webfonts[ $slug ];
+		unset( $this->registered_webfonts[ $slug ] );
+	}
+
+	/**
+	 * Get the font slug.
+	 *
+	 * @param array|string $to_convert The value to convert into a slug. Expected as the web font's array or a font-family as a string.
+	 */
+	public static function get_font_slug( $to_convert ) {
+		if ( is_array( $to_convert ) ) {
+			if ( isset( $to_convert['font-family'] ) ) {
+				$to_convert = $to_convert['font-family'];
+			} elseif ( isset( $to_convert['fontFamily'] ) ) {
+				$to_convert = $to_convert['fontFamily'];
+			} else {
+				_doing_it_wrong( __FUNCTION__, __( 'Could not determine the font family name.' ), '6.0.0' );
+				return false;
+			}
+		}
+
+		return sanitize_title( $to_convert );
```

Comment:
How will this handle the same font families registered by different providers? (A local provider font from the theme and a Google Font from a plugin, for example)

Reply:
It will register both, for now. We're thinking of creating a separate issue so we can handle font face conflicts in a better way. One option is to introduce a `WP_Webfont` class which would encapsulate a font face declaration. This class would have a method like `is_equal` which would be called by the registry to determine whether to override or to add the font face to the list.